ytypes/leaf.go

```diff
@@ -902,7 +902,7 @@ func sanitizeGNMI(parent interface{}, schema *yang.Entry, fieldName string, tv *
 func gNMIToYANGTypeMatches(ykind yang.TypeKind, tv *gpb.TypedValue) bool {
 	var ok bool
 	switch ykind {
-	case yang.Ybool:
+	case yang.Ybool, yang.Yempty:
```

Comment:
I suspect that we really need to make `yang.Yempty` always set the field to `true` if we receive this. Since `YANGEmpty` isn't a pointer in the generated code, then if we receive an empty field that has a bool value of `false` then this won't be something that the client application can tell was received. Could you add this handling as a special case please?

Reply:
Actually there already appears to be handling for this case:
https://github.com/openconfig/ygot/blob/f01913aa54675e818f97bf5a55692755e2b6ff10/util/reflect.go#L256-L262

When unmarshalling an empty leaf from a `TypedValue` in the current code, `unmarshalScalar` returns the non-pointer bool value, which `unmarshalLeaf` then takes and then feeds into `InsertIntoStruct` (directly or indirectly) to reach this special handling code.

This change doesn't support unmarshalling empty leafs (either JSON or gNMI) into a union type. However given its complexity I think it deserves to be addressed separately.

Reply:
I explored the question of unions containing empty in #582.